content/browser/renderer_host/brave_navigation_controller_impl.cc

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/content/browser/renderer_host/brave_navigation_controller_impl.h"
+
+#include "brave/components/constants/webui_url_constants.h"
+#include "build/build_config.h"
+#include "content/public/common/url_constants.h"
+
+namespace content {
+
+BraveNavigationControllerImpl::BraveNavigationControllerImpl(
+    BrowserContext* browser_context,
+    FrameTree& frame_tree,
+    NavigationControllerDelegate* delegate)
+    : NavigationControllerImpl(browser_context, frame_tree, delegate) {}
+
+BraveNavigationControllerImpl::~BraveNavigationControllerImpl() {}
+
+base::WeakPtr<NavigationHandle>
+BraveNavigationControllerImpl::LoadURLWithParams(const LoadURLParams& params) {
```

Comment:
do we really need to create `BraveNavigationControllerImpl` for the sole purpose of preventing `chrome-untrusted://chat` opening? To me this looks like a wrong approach. This way we had to block/ban all other unwanted pages, but we don't do that.

Maybe we can fix the chat WebUI to be no-op if it's opened in an unsupported context?

Reply:
>do we really need to create BraveNavigationControllerImpl for the sole purpose of preventing chrome-untrusted://chat opening?

this is the best place I could find for such preventing. 
I was surprised by the fact there were no any brave-related overrides or patches around `NavigationControllerImpl`, but nevertheless I had put the change there.

>Maybe we can fix the chat WebUI to be no-op if it's opened in an unsupported context?

Basically the first commit afeef6f90d9028b5e007c4082d9d849d6189afaa in the PR  fixes the crash itself at WebUI, but doesn't prevent the page from load.
I will recheck if the page transition parameter is available at this place, but as far as I remember, it doesn't.


Thanks for the feedback.

Reply:
I may try to use 
[WebContentsObserver](https://github.com/brave/brave-core/blob/master/browser/ui/webui/ai_chat/ai_chat_ui_page_handler.h#L79)
[DidStartNavigation](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/web_contents_observer.h;drc=42f0b9c4f60f67676d9096b3e94df9fd47f568f4;bpv=1;bpt=1;l=307?q=WebContentsObserver&ss=chromium%2Fchromium%2Fsrc)
[GetPageTransition](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/navigation_handle.h;l=269;drc=445f8cfdbb63b29ce44f728f6b5015cf18df5534)

to detect the unwanted navigations and then find the way to prevent them.

Reply:
Thanks, changed the PR to use `NavigationThrottle`
https://github.com/brave/brave-core/pull/20996/commits/025abf5e0bfe6df6297eb4de3f109f839add8239

Reply:
> Thanks, changed the PR to use `NavigationThrottle` [025abf5](https://github.com/brave/brave-core/commit/025abf5e0bfe6df6297eb4de3f109f839add8239)

much cleaner, thanks!